File: Algorithm/python/Eratosthenes.py
def eratosthenes(n):
    era = [True for _ in range(n+1)]
    era[0], era[1] = False, False
    primes = [] #prime 배열 안쓴다면 생략 가능
    for i in range(2, int(n**0.5)+1): # 제곱근까지만 봐도 무방! 단, prime 배열 쓴다면 n+1 까지
        if era[i]:
            primes.append(i) #prime 배열 안쓴다면 생략 가능
            for j in range(i*2, n+1, i):
                era[j] = False
    return era, primes # 둘 중 필요한 것 골라 쓰면 된다.

#소수 판별 함수
def isPrime(n):
    if n<=1:
        return False
    for i in range(2, int(n**0.5)+1):
        if n%i==0:
            return False
    return True

File: Algorithm/cpp/Eratosthenes.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int n;
vector<bool> era;
vector<int> primes;
void eratosthenes(int n){
    era[0]=era[1]=false;
    
    for (int i=2; i<=n; i++){ //primes 배열 쓴다면 n까지!
    //for (int i=2; i<=sqrt(n); i++){ // 안쓴다면 sqrt(n)까지만 해도 ok.
        if (era[i]){
            primes.push_back(i); //이거도 안쓰면 필요 없음.
            for (int j=i*2; j<n+1; j+=i){
                era[j] = false;
            }
        }
    }
}

int main() {
    n = 100;
    era = vector<bool>(n+1, true);
    eratosthenes(n);
}